crates/router/src/core/webhooks/webhook_events.rs

```diff
@@ -153,10 +155,108 @@ pub async fn list_delivery_attempts(
     }
 }
 
+#[instrument(skip(state))]
+pub async fn retry_delivery_attempt(
+    state: AppState,
+    merchant_id_or_profile_id: String,
+    event_id: String,
+) -> RouterResponse<api::webhook_events::EventRetrieveResponse> {
+    let store = state.store.as_ref();
+
+    let (account, key_store) =
+        determine_identifier_and_get_key_store(state.clone(), merchant_id_or_profile_id).await?;
+
+    let event_to_retry = store
+        .find_event_by_merchant_id_event_id(&key_store.merchant_id, &event_id, &key_store)
+        .await
+        .to_not_found_response(errors::ApiErrorResponse::EventNotFound)?;
+
+    let business_profile = match account {
+        MerchantAccountOrBusinessProfile::MerchantAccount(_) => {
+            let business_profile_id = event_to_retry
+                .business_profile_id
+                .get_required_value("business_profile_id")
```

Comment:
Shouldn't we check for profile_id in `merchant_account.default_profile_id` ?

Reply:
If the `business_profile_id` is not present in the event, then we're trying to retry a very old event which does not have many of the fields populated, which I don't want to support. We also need the `request` field to be populated, to be able to read the request body and headers sent previously and retry the webhook.

Moreover, if there are multiple business profiles associated with the merchant account and the default profile ID is `profile1`, but this event was related to `profile2`, then we'd end up populating incorrect data by populating `business_profile_id` as `profile1` in the database.